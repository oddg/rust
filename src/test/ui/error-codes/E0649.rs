enum E {
    A = 0,
}

impl Drop for E {
    fn drop(&mut self) {
        println!("Drop");
    }
}

fn main() {
    let e = E::A;
    let i = e as u32; //~ ERROR E0649
}
