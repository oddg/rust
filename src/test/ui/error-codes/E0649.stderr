error[E0649]: cannot cast `enum` implementing `Drop` `E` to integer `u32`
  --> $DIR/E0649.rs:13:13
   |
LL |     let i = e as u32;
   |             ^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0649`.
